jupyterhub_singleuser_profiles/service.py

```diff
@@ -37,91 +41,110 @@ def __init__(self, server_url, token, namespace=None, verify_ssl=True):
     self.k8s_api_instance = client.CoreV1Api(k8s_client)
     self.os_client = DynamicClient(k8s_client)
 
-  def get_template(self, name):
-    tmplt = self.os_client.resources.get(api_version='template.openshift.io/v1', kind='Template')
-    if isinstance(tmplt, list):
-      tmplt = tmplt[0] # Openshift returns list of 2 resource types, we want template.openshift.io/v1/processedtemplates
+  def get_service_reference_config_map (self, user):
+    _LOGGER.debug("Getting referece config map")
+    config_map_wrapper = self.os_client.resources.get(api_version='v1', kind='ConfigMap')
+    try:
+      body = {
+          'kind': 'ConfigMap',
+          'apiVersion': 'v1',
+          'metadata': {'name': _REFERENCE_CM_NAME %(escape(user))},
+          'data': {}
+        }
+      config_map_wrapper.create(body=body, namespace=self.namespace)
+      result = config_map_wrapper.get(namespace=self.namespace, name=_REFERENCE_CM_NAME %(escape(user)))
```

Comment:
I would suggest to remove this `get` and change the exception handling to 
```
if e.status != 409:
  _LOGGER.error(...)
  raise

... wrapper.get(...
```

i.e. use the exception for error handling and do the get outside of it